packages/x-components/src/x-modules/tagging/wiring.ts

```diff
@@ -76,12 +77,22 @@ export const setSessionDuration = wireCommit('setSessionDuration');
  *
  * @public
  */
-export const trackWire = moduleDebounce(
+export const trackQueryWire = moduleDebounce(
   wireDispatch('track'),
   ({ state }) => state.config.queryTaggingDebounceMs,
   'UserClearedQuery'
 );
 
+/**
+ * Tracks the tagging of the result if the user clicks on it.
+ *
+ * @public
+ */
+export const trackResultClickedWire = mapWire(
+  wireDispatch('track'),
+  ({ tagging }: Result) => tagging.click!
```

Comment:
We're lying to Typescript here. We're assuring typescript that `tagging.click` will be non-null when it actually can be `undefined`, and then handling it inside the `track` action. 

Remove the `!` and change the type of the action so it accept possible `undefined` values.

Reply:
After talking with Ivan, the responsibility of emitting the event goes to the BaseResultLink component, so we need to emit the event only if there is a payload.

I'll keep like this and remove the test about this issue.